### PR TITLE
Feat/182 import candidates

### PR DESCRIPTION
**Pull request by joonatanaatos**
## WHY:

Fixes [Issue link]

## Check off each of the following tasks as they are completed

- [x] I have reviewed the changes myself in this PR. Please check the [self-review document](https://github.com/OpenVAA/voting-advice-application/blob/main/docs/contributing/self-review.md)
- [ ] I have added or edited unit tests.
- [ ] I have run the unit tests successfully.
- [x] I have tested this change on my own device.
- [ ] I have tested this change on other devices (Using Browserstack is recommended).
- [ ] I have tested my changes using the [WAVE extension](https://wave.webaim.org/extension/)
- [x] I have added documentation where necessary.
- [x] Is there an existing issue linked to this PR?

**Clean up your git commit history before submitting the pull request!**

### CODE REVIEW

**Review by TrixterTheTux — CHANGES_REQUESTED** [review]
Please add the relevant documentation for downloading (or updating) the submodules after the repository is cloned

File: docs/candidate-app/import-functionality.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ - `constituency` (id)
+ - `candidate` (id)
+ - `party` (id)
+ - `electionSymbol` (text)
+ - `published` (true / false)
+
+The composite of the fields `election`, `constituency`, `candidate` and `party` is used as the unique identifier for nominations.
+
+The import will fail if one or more of the fields are not defined.
+
```
Comment (TrixterTheTux) [inline docs/candidate-app/import-functionality.md:32]: I would consider adding an example CSV for both here (inside code blocks, i.e. ```), so it's a bit more clear and faster to get started using the import functionality

File: docs/docker-setup-guide.md [hunk 2]
```diff
@@ -13,7 +13,13 @@ backend into a single image that can be built with just a few commands.
 ## Getting started
 
 To build the development Docker image, go to the project root folder make a copy of the `.env.example` file and rename 
-the copy as `.env`. Then run `yarn install` and `yarn dev` or `npm install` and `npm run dev` in the project root folder. 
+the copy as `.env`.
+
+Then you need to install the git submodules by running:
+1. `git submodule init`
+2. `git submodule update`
```
Comment (TrixterTheTux) [inline docs/docker-setup-guide.md:20]: These steps can be shortened as `git submodule update --init`, it's also a good habit to add `--recursive`, so the single command to run could be `git submodule update --init --recursive` instead.

File: docs/docker-setup-guide.md [hunk 3]
```diff
@@ -13,7 +13,13 @@ backend into a single image that can be built with just a few commands.
...
+
+Finally, run `yarn install` and `yarn dev` or `npm install` and `npm run dev` in the project root folder.
```
Comment (TrixterTheTux) [inline docs/docker-setup-guide.md:22]: I would reword this as it's a bit weird to read due to the use of "A and B or C and D" (though this also isn't scoped to this PR itself)

**Review by TrixterTheTux — APPROVED** [review]

File: backend/vaa-strapi/package.json [hunk 4]
```diff
@@ -27,7 +27,8 @@
     "@strapi/provider-email-nodemailer": "^4.15.5",
     "@strapi/strapi": "4.11.7",
     "better-sqlite3": "7.4.6",
-    "pg": "^8.8.0"
+    "pg": "^8.8.0",
+    "strapi-plugin-import-export-entries": "file:./strapi-plugin-import-export-entries"
```
Comment (johannalaitila) [inline backend/vaa-strapi/package.json:31]: Question: why are we importing this as a submodule and installing it in Dockerfile instead of having it as an normal npm package?

File: backend/vaa-strapi/src/api/candidate/content-types/candidate/lifecycles.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import crypto from 'crypto';
+
+export default {
+  beforeCreate(event) {
+    event.params.data.registrationKey = crypto.randomUUID();
```
Comment (TrixterTheTux) [inline backend/vaa-strapi/src/api/candidate/content-types/candidate/lifecycles.ts:5]: Maybe worth testing, but this could also overwrite the explicitly configured registration key when importing (perhaps this should be `event.params.data.registrationKey ?? crypto.randomUUID()` instead if we're not able to overwrite this?)

**Review by TrixterTheTux — APPROVED** [review]

**Review by eemelioksanen — APPROVED** [review]
Looks good!
